File: 83.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode *cur=head;
        ListNode *ans=head;
        if(head==NULL) return NULL;
        while(cur->next!=NULL)
        {
            cur=cur->next;
            if(cur->val==head->val) continue;
            else
            {
                head->next=cur;
                head=head->next;
            }
        }
        head->next=NULL;
        return ans;
    }
};

File: 83.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cur=ans=head
        if head==None: return head
        while cur.next:
            cur=cur.next
            if head.val==cur.val: continue
            else:
                head.next=cur
                head=head.next
        head.next=None
        return ans
